examples/chef/common/clusters/door-lock/chef-lock-manager.cpp

```diff
@@ -0,0 +1,376 @@
+                     "Unable to get number of credentials supported per user when initializing lock endpoint, defaulting to 5 "
+                     "[endpointId=%d]",
+                     endpointId);
+        numberOfCredentialsSupportedPerUser = 5;
+    }
+
+    uint8_t numberOfWeekDaySchedulesPerUser = 0;
+    if (!DoorLockServer::Instance().GetNumberOfWeekDaySchedulesPerUserSupported(endpointId, numberOfWeekDaySchedulesPerUser))
+    {
+        ChipLogError(Zcl,
+                     "Unable to get number of supported week day schedules per user when initializing lock endpoint, defaulting to "
+                     "10 [endpointId=%d]",
+                     endpointId);
+        numberOfWeekDaySchedulesPerUser = 10;
+    }
+
+    uint8_t numberOfYearDaySchedulesPerUser = 0;
+    if (!DoorLockServer::Instance().GetNumberOfYearDaySchedulesPerUserSupported(endpointId, numberOfYearDaySchedulesPerUser))
+    {
+        ChipLogError(Zcl,
+                     "Unable to get number of supported year day schedules per user when initializing lock endpoint, defaulting to "
+                     "10 [endpointId=%d]",
+                     endpointId);
+        numberOfYearDaySchedulesPerUser = 10;
+    }
+
+    uint8_t numberOfHolidaySchedules = 0;
+    if (!DoorLockServer::Instance().GetNumberOfHolidaySchedulesSupported(endpointId, numberOfHolidaySchedules))
+    {
+        ChipLogError(
+            Zcl,
+            "Unable to get number of supported holiday schedules when initializing lock endpoint, defaulting to 10 [endpointId=%d]",
+            endpointId);
+        numberOfHolidaySchedules = 10;
+    }
+
+    mEndpoints.emplace_back(endpointId, numberOfSupportedUsers, numberOfSupportedCredentials, numberOfWeekDaySchedulesPerUser,
+                            numberOfYearDaySchedulesPerUser, numberOfCredentialsSupportedPerUser, numberOfHolidaySchedules);
+
+    // Refer to 5.2.10.34. SetUser Command, when Creat a new user record
+    //    - UserIndex value SHALL be set to a user record with UserType set to Available
+    //    - UserName MAY be null causing new user record to use empty string for UserName
+    //        otherwise UserName SHALL be set to the value provided in the new user record.
+    //    - UserUniqueID MAY be null causing new user record to use 0xFFFFFFFF for UserUniqueID
+    //        otherwise UserUniqueID SHALL be set to the value provided in the new user record
+    //    - UserStatus MAY be null causing new user record to use OccupiedEnabled for UserStatus
+    //        otherwise UserStatus SHALL be set to the value provided in the new user record
+    //   - UserType MAY be null causing new user record to use UnrestrictedUser for UserType
+    //        otherwise UserType SHALL be set to the value provided in the new user record.
+    uint16_t userIndex(1);
+    chip::FabricIndex creator(1);
+    chip::FabricIndex modifier(1);
+    const chip::CharSpan userName = chip::CharSpan::fromCharString("user1"); // default
```

Comment:
In the future, please use "user1"_span or so.

Reply:
Got it. Thanks for the advice.